### ISSUE TITLE
Setup issue during prereq installation via requirements.txt

### ISSUE THREAD

**Issue by asifaaag** [vmware/vsphere-automation-sdk-python#36]
I'm hitting issues on CentOS and IOSX, while trying to install the prereqs from the requirements.txt:

[00:43][asulikeri:~/Documents/vsphere-automation-sdk-python-master]$ source SDKDemo/bin/activate
(SDKDemo) [00:44][asulikeri:~/Documents/vsphere-automation-sdk-python-master]$ pip3 install -r requirements.txt
Ignoring suds: markers ‘python_version =6.5 (from -r requirements.txt (line 1))
Downloading pyvmomi-6.5.0.2017.5-1.tar.gz (252kB)
100% |████████████████████████████████| 256kB 1.8MB/s
Collecting suds-jurko (from -r requirements.txt (line 3))
Downloading suds-jurko-0.6.tar.bz2 (143kB)
100% |████████████████████████████████| 153kB 3.8MB/s
Collecting vapi-client-bindings (from -r requirements.txt (line 4))
Could not find a version that satisfies the requirement vapi-client-bindings (from -r requirements.txt (line 4)) (from versions: )
No matching distribution found for vapi-client-bindings (from -r requirements.txt (line 4))

And after that when I try to run the script I hit the following issues:

(SDKDemo) [00:44][asulikeri:~/Documents/vsphere-automation-sdk-python-master]$ python samples/vsphere/vcenter/setup/main.py
Traceback (most recent call last):
File “samples/vsphere/vcenter/setup/main.py”, line 25, in
import pyVim.connect
ModuleNotFoundError: No module named ‘pyVim’

Any idea if I’m missing out on anything.

**Comment by sushiman** [comment]
Hi @asifaaag , I stumbled today on the same issue.  To solve the problem, git clone the repository in your working dir and then execute pip again.

> git clone https://github.com/vmware/vsphere-automation-sdk-python
> cd vsphere-automation-sdk-python
> pip install -r requirements.txt --extra-index-url **<absolute path of your working dir>**\vsphere-automation-sdk-python\lib\`

Besides, I also stumbled on another issue with the default pyOpenSSL installed version after trying to connect to a vCenter. Just uninstall pyOpenSSL and install this version:

> pip uninstall pyOpenSSL
> pip install pyOpenSSL==16.2.0

**Comment by asifaaag** [comment]
Great that worked, thanks a lot sushiman.

**Comment by asifaaag** [comment]
This is strange. This is not directly resolved. Looks like I'll have to individually extract all the zips of bindings and runtimes and then add them to sys.path and only after that the pip install -r requirements.txt works.

**Comment by tianhao64** [comment]
adding --extra-index-url will install the package in lib folder automatically. You shouldn't have to do unzip the package separately.

**Comment by asifaaag** [comment]
Yes @tianhao64 thats what I have tried, here are the precise steps:
git clone https://github.com/vmware/vsphere-automation-sdk-python
cd vsphere-automation-sdk-python/
[root@vmc-centos vsphere-automation-sdk-python]# pip install -r requirements.txt --extra-index-url ~/loadutil2/vsphere-automation-sdk-python/lib/
Ignoring suds-jurko: markers 'python_version >= "3.0"' don't match your environment
Collecting pyVmomi>=6.5 (from -r requirements.txt (line 1))
  Url '/root/loadutil2/vsphere-automation-sdk-python/lib/pyvmomi/' is ignored. It is either a non-existing path or lacks a specific scheme.
Collecting suds (from -r requirements.txt (line 2))
  Url '/root/loadutil2/vsphere-automation-sdk-python/lib/suds/' is ignored. It is either a non-existing path or lacks a specific scheme.
  Using cached suds-0.4.tar.gz
Collecting vapi-client-bindings (from -r requirements.txt (line 4))
  Could not find a version that satisfies the requirement vapi-client-bindings (from -r requirements.txt (line 4)) (from versions: )
No matching distribution found for vapi-client-bindings (from -r requirements.txt (line 4))

**Comment by dungla2011** [comment]
I have the same error, and can not fix:

```
[root@galaxycloud vsphere-automation-sdk-python]# pip install -r requirements.txt --extra-index-url /share/vsphere-automation-sdk-python/lib/
Ignoring suds: markers 'python_version < "3"' don't match your environment
Looking in indexes: https://pypi.org/simple, /share/vsphere-automation-sdk-python/lib/
Collecting pyVmomi>=6.7 (from -r requirements.txt (line 1))
  Url '/share/vsphere-automation-sdk-python/lib/pyvmomi/' is ignored. It is either a non-existing path or lacks a specific scheme.
  Using cached https://files.pythonhosted.org/packages/ee/9a/7b24db4501e720fbd0a3578a8dde4b03d20b6fc3427e992dce19a3cf5cfc/pyvmomi-6.7.1.zip
Collecting suds-jurko (from -r requirements.txt (line 3))
  Url '/share/vsphere-automation-sdk-python/lib/suds-jurko/' is ignored. It is either a non-existing path or lacks a specific scheme.
  Using cached https://files.pythonhosted.org/packages/bd/6f/54fbf0999a606680d27c69b1ad12dfff62768ecb9fe48524cebda6eb4423/suds-jurko-0.6.tar.bz2
Collecting tabulate (from -r requirements.txt (line 4))
  Url '/share/vsphere-automation-sdk-python/lib/tabulate/' is ignored. It is either a non-existing path or lacks a specific scheme.
  Using cached https://files.pythonhosted.org/packages/12/c2/11d6845db5edf1295bc08b2f488cf5937806586afe42936c3f34c097ebdc/tabulate-0.8.2.tar.gz
Collecting vapi-client-bindings==1.5.0 (from -r requirements.txt (line 5))
  Could not find a version that satisfies the requirement vapi-client-bindings==1.5.0 (from -r requirements.txt (line 5)) (from versions: 2.5.0)
No matching distribution found for vapi-client-bindings==1.5.0 (from -r requirements.txt (line 5))
You have mail in /var/spool/mail/root
```

**Requirements.txt:**
```
pyVmomi >= 6.7
suds ; python_version < '3'
suds-jurko ; python_version >= '3.0'
tabulate
vapi-client-bindings == 1.5.0
vmc-client-bindings
vapi-vmc-client
nsx-policy-python-sdk
vmc-app-python-sdk 
```

```
[root@galaxycloud vsphere-automation-sdk-python]# python --version
Python 3.6.3
```

**Comment by tianhao64** [comment]
@asifaaag @dungla2011 Please give full file path to the lib folder. such as
pip install -r requirements.txt --extra-index-url file:///share/vsphere-automation-sdk-python/lib/
The sample cmd is listed in the main README quick start section.

**Comment by tianhao64** [comment]
BTW, --upgrade --force-reinstall is also required. Please see README for examples:
https://github.com/vmware/vsphere-automation-sdk-python#installing-required-python-packages

**Comment by divyathallamudi** [comment]
Hi, I have the same error and cannot fix.
Running from Windows:

$ pip install --upgrade --force-reinstall -r requirements.txt --extra-index-url file://`pwd`/lib
Ignoring suds: markers 'python_version < "3"' don't match your environment
Looking in indexes: https://pypi.org/simple, file:///c/Users/dthallam/Documents/GitHub/vsphere-automation-sdk-python/lib
Collecting pyVmomi>=6.7 (from -r requirements.txt (line 1))
  Url 'file:///c/Users/dthallam/Documents/GitHub/vsphere-automation-sdk-python/lib/pyvmomi/' is ignored: it is neither a file nor a directory.
  Using cached https://files.pythonhosted.org/packages/71/24/0bb1257b3bc89f7b2facdbad91cc56902d116d649a263c242ef32f73110e/pyvmomi-6.7.1.2018.12.zip
Collecting suds-jurko (from -r requirements.txt (line 3))
  Url 'file:///c/Users/dthallam/Documents/GitHub/vsphere-automation-sdk-python/lib/suds-jurko/' is ignored: it is neither a file nor a directory.
  Using cached https://files.pythonhosted.org/packages/bd/6f/54fbf0999a606680d27c69b1ad12dfff62768ecb9fe48524cebda6eb4423/suds-jurko-0.6.tar.bz2
Collecting vapi-client-bindings==2.5.0 (from -r requirements.txt (line 4))
  Url 'file:///c/Users/dthallam/Documents/GitHub/vsphere-automation-sdk-python/lib/vapi-client-bindings/' is ignored: it is neither a file nor a directory.
  Using cached https://files.pythonhosted.org/packages/bc/44/0643dfb733f6300ca02c1488c70e997d248f3d60b2f9e76e899c0b8ead70/vapi_client_bindings-2.5.0.tar.gz
Collecting vmc-client-bindings (from -r requirements.txt (line 5))
  Url 'file:///c/Users/dthallam/Documents/GitHub/vsphere-automation-sdk-python/lib/vmc-client-bindings/' is ignored: it is neither a file nor a directory.
  **Could not find a version that satisfies the requirement vmc-client-bindings (from -r requirements.txt (line 5)) (from versions: )
No matching distribution found for vmc-client-bindings (from -r requirements.txt (line 5))**

Python version:
Python 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 14:05:16) [MSC v.1915 32 bit (Intel)] on win32

Requirements file:

pyVmomi >= 6.7
suds ; python_version < '3'
suds-jurko ; python_version >= '3.0'
vapi-client-bindings == 2.5.0
vmc-client-bindings
vapi-vmc-client
nsx-policy-python-sdk
vmc-app-python-sdk

**Comment by tianhao64** [comment]
@divyathallamudi Can you follow the Windows example listed here?: https://github.com/vmware/vsphere-automation-sdk-python#installing-required-python-packages

**Comment by tianhao64** [comment]
@divyathallamudi You need one more "/". This should work:
pip install --upgrade --force-reinstall -r requirements.txt --extra-index-url file:\\\C:\Users\strefethen\github\vsphere-automation-sdk-python\lib

**Comment by tianhao64** [comment]
@divyathallamudi "Url 'file:\C:Usersdthallamdocumentsgithubvsphere-automation-sdk-pythonlib/pyvmomi/' is ignored: it is neither a file nor a directory." this is weird. Looks like the file path was incorrectly escaped. Not sure why. Are you still using Windows? Cmd Prompt or Powershell?

Was there three / following the file: or two? The cmd should be(3 slashes):
`pip install --upgrade --force-reinstall -r requirements.txt --extra-index-url file:\\\C:\Users\dthallam\documents\github\vsphere-automation-sdk-python\lib`

**Comment by divyathallamudi** [comment]
@tianhao64 I have installed the package directly to work it out but tried per your recommendation in a virtual environment and its working fine!

Steps performed:
1. Changed the value for "vapi-client-bindings" in requirements.txt to 2.5.0
vapi-client-bindings == 2.5.0
2. Ran the command pip install --upgrade --force-reinstall -r requirements.txt --extra-index-url file:\\C:\Users\dthallam\documents\github\vsphere-automation-sdk-python\lib

Appreciate your response and help here!

**Comment by pavanshivakumar** [comment]
python samples/vsphere/vcenter/vm/list_vms.py -s <ip> -u <username> -p <password> -v
results in the error requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://<ip>/api

Any help on the above would be appreciated

**Comment by yildirima** [comment]
> BTW, --upgrade --force-reinstall is also required. Please see README for examples:
> https://github.com/vmware/vsphere-automation-sdk-python#installing-required-python-packages

Thanks. Fixed my problem.
